### PR TITLE
Remove dependency on Evm Address in wormhole prover to support Solana

### PR DESCRIPTION
**Pull request by kiseln**

### CODE REVIEW

File: near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs [hunk 1]
```diff
@@ -199,7 +199,7 @@ impl TryInto<DeployTokenMessage> for ParsedVAA {
 
         Ok(DeployTokenMessage {
             token: transfer.token.parse().map_err(stringify)?,
-            token_address: to_omni_address(self.emitter_chain, &transfer.token_address.0),
+            token_address: transfer.token_address,
             emitter_address: to_omni_address(self.emitter_chain, &self.emitter_address),
```
Comment (karim-en) [inline near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs:203]: Let's avoid maintaining the `to_omni_address` with the `self.emitter_chain`, and get the chain kind from the `token_address`

File: near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs [hunk 2]
```diff
@@ -115,13 +115,13 @@ impl ParsedVAA {
...
     token: String,
-    token_address: EvmAddress,
+    token_address: OmniAddress,
 }
 
 #[derive(Debug, BorshDeserialize)]
 struct FinTransferWh {
     payload_type: ProofKind,
-    _token: String,
+    _token: OmniAddress,
```
Comment (karim-en) [inline near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs:124]:
```suggestion
    token_address: OmniAddress,
```

**Review by karim-en — CHANGES_REQUESTED** [review]

File: near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs (hunk 1 above)
Comment (kiseln) [reply near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs]: 0c2510eeafab68f765a2c1bd5ed51ad1a0db42b8

File: near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs (hunk 2 above)
Comment (kiseln) [reply near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs]: 0c2510eeafab68f765a2c1bd5ed51ad1a0db42b8

File: near/omni-types/src/lib.rs [hunk 3]
```diff
@@ -162,10 +162,22 @@ impl OmniAddress {
     pub fn from_evm_address(chain_kind: ChainKind, address: EvmAddress) -> Result<Self, String> {
         match chain_kind {
             ChainKind::Eth => Ok(Self::Eth(address)),
+            ChainKind::Arb => Ok(Self::Arb(address)),
+            ChainKind::Base => Ok(Self::Base(address)),
             _ => Err(format!("{chain_kind:?} is not an EVM chain")),
         }
     }
 
+    pub fn from_bytearray(chain_kind: ChainKind, address: &[u8]) -> Result<Self, String> {
```
Comment (karim-en) [inline near/omni-types/src/lib.rs:171]:
```suggestion
    pub fn new_from_slice(address: &[u8], chain_kind: ChainKind) -> Result<Self, String> {
```

**Review by karim-en — CHANGES_REQUESTED** [review]

File: near/omni-types/src/lib.rs [hunk 4]
```diff
@@ -162,10 +162,22 @@ impl OmniAddress {
     pub fn from_evm_address(chain_kind: ChainKind, address: EvmAddress) -> Result<Self, String> {
```
Comment (karim-en) [inline near/omni-types/src/lib.rs:162]:
```suggestion
    pub fn new_from_evm_address(chain_kind: ChainKind, address: EvmAddress) -> Result<Self, String> {
```

**Review by karim-en — APPROVED** [review]

**Pull request merged**